### PR TITLE
Add loaded components popup to dev-info

### LINKED ISSUE #1347: Show loaded components in dev-info
`hass.config.core.components` contains a list of all loaded components and platforms. We should print those in the dev-info page so that users can easily report it in issues.

### PR DESCRIPTION
**Pull request by cgarwood**
Adds a link to dev-info that pops up a list of loaded components from hass.config.components.

Fixes #1347 

image
image

### CODE REVIEW

File: src/panels/dev-info/ha-panel-dev-info.js [hunk 1]
```diff
@@ -220,6 +221,18 @@ class HaPanelDevInfo extends PolymerElement {
         </template>
       </paper-dialog-scrollable>
     </paper-dialog>
+
+    <paper-dialog with-backdrop id="showcomponents">
```
Comment (balloob) [inline src/panels/dev-info/ha-panel-dev-info.js]: This is not how we do dialogs anymore. Instead, use this https://github.com/home-assistant/home-assistant-polymer/blob/master/src/panels/config/config-entries/ha-config-entries.js#L128-L133

Comment (balloob) [reply src/panels/dev-info/ha-panel-dev-info.js]: And if you want, feel free to put a PR for the dev docs too as it's clearly missing a piece about dialogs.

**Comment by cgarwood** [comment]
Tried to copy the dialog setup form the config-entry example and stripped out the extra attributes/functions that this dialog doesn't need. Let me know if I missed something.

**Review by balloob — APPROVED** [review]

**Comment by balloob** [comment]
Perfect.

**Pull request merged**